boards/sodaq-sara-sff/include/periph_conf.h

```diff
@@ -0,0 +1,152 @@
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     boards_sodaq-sara-sff
+ * @{
+ *
+ * @file
+ * @brief       Configuration of CPU peripherals for the SODAQ SARA SFF board
+ *
+ * @author      Kees Bakker <kees@sodaq.com>
+ */
+
+#ifndef PERIPH_CONF_H
+#define PERIPH_CONF_H
+
+#include <stdint.h>
+
+#include "cpu.h"
+#include "periph_cpu.h"
+#include "cfg_clock_default.h"
+#include "cfg_rtc_default.h"
+#include "cfg_rtt_default.h"
+#include "cfg_timer_default.h"
+#include "cfg_usbdev_default.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @name UART configuration
+ * @{
+ * See Table 7-1 of the SAM D21 Datasheet (p. 29)
+ */
+static const uart_conf_t uart_config[] = {
+    {
+        .dev = &SERCOM5->USART,
+        .rx_pin = GPIO_PIN(PB, 3),  /* AIN11 */
+        .tx_pin = GPIO_PIN(PB, 2),  /* AIN10 */
+        .mux = GPIO_MUX_D,
+        .rx_pad = UART_PAD_RX_1,
+        .tx_pad = UART_PAD_TX_0,
+        .flags = UART_FLAG_NONE,
+        .gclk_src = SAM0_GCLK_MAIN,
+    },
+    {
+        /* Connected to the UBlox */
+        .dev = &SERCOM2->USART,
+        .rx_pin = GPIO_PIN(PA, 13),
+        .tx_pin = GPIO_PIN(PA, 12),
+        .mux = GPIO_MUX_C,
+        .rx_pad = UART_PAD_RX_1,
+        .tx_pad = UART_PAD_TX_0,
+        .flags = UART_FLAG_NONE,
+        .gclk_src = SAM0_GCLK_MAIN,
+    },
+};
+
+/* interrupt function name mapping */
+#define UART_0_ISR          isr_sercom5
+#define UART_1_ISR          isr_sercom2
+
+#define UART_NUMOF          ARRAY_SIZE(uart_config)
+/** @} */
+
+/**
+ * @name ADC configuration
+ * @{
+ */
+
+/* ADC Default values */
+#define ADC_PRESCALER                       ADC_CTRLB_PRESCALER_DIV512
+
+#define ADC_NEG_INPUT                       ADC_INPUTCTRL_MUXNEG_GND
+#define ADC_GAIN_FACTOR_DEFAULT             ADC_INPUTCTRL_GAIN_DIV2
+#define ADC_REF_DEFAULT                     ADC_REFCTRL_REFSEL_INTVCC1
+
+static const adc_conf_chan_t adc_channels[] = {
+    /* port, pin, muxpos */
+    { GPIO_PIN(PA, 2), ADC_INPUTCTRL_MUXPOS_PIN0 },     /* A0 */
+    { GPIO_PIN(PA, 3), ADC_INPUTCTRL_MUXPOS_PIN1 },     /* A1 */
+    { GPIO_PIN(PB, 8), ADC_INPUTCTRL_MUXPOS_PIN2 },     /* A2 */
+    { GPIO_PIN(PB, 9), ADC_INPUTCTRL_MUXPOS_PIN3 },     /* A3 */
+    { GPIO_PIN(PA, 6), ADC_INPUTCTRL_MUXPOS_PIN6 },     /* A4 */
+    { GPIO_PIN(PA, 7), ADC_INPUTCTRL_MUXPOS_PIN7 },     /* A5 */
+    { GPIO_PIN(PA, 8), ADC_INPUTCTRL_MUXPOS_PIN16 },    /* A6 */
+    { GPIO_PIN(PA, 9), ADC_INPUTCTRL_MUXPOS_PIN17 },    /* A7 */
+    { GPIO_PIN(PA, 10), ADC_INPUTCTRL_MUXPOS_PIN18 },   /* A8 */
+    { GPIO_PIN(PA, 11), ADC_INPUTCTRL_MUXPOS_PIN19 },   /* A9 */
+#if 0
+    /* These pins are also used for RX/TX uart0 */
+    { GPIO_PIN(PB, 2), ADC_INPUTCTRL_MUXPOS_PIN10 },    /* A10 */
+    { GPIO_PIN(PB, 3), ADC_INPUTCTRL_MUXPOS_PIN11 },    /* A11 */
+#endif
```

Comment:
dito

Reply:
These is more or less to document that the pins are multi purpose. By default the are used by the UART, so they can't be enabled for ADC.